hadoop-hdfs-project/hadoop-hdfs-client/src/main/java/org/apache/hadoop/hdfs/web/WebHdfsFileSystem.java

```diff
@@ -846,9 +846,27 @@ private T runWithRetry() throws IOException {
             node = url.getAuthority();
           }
           try {
-            IOException newIoe = ioe.getClass().getConstructor(String.class)
-                .newInstance(node + ": " + ioe.getMessage());
-            newIoe.initCause(ioe.getCause());
+            final Throwable cause = ioe.getCause();
+            IOException newIoe = null;
+            if (cause != null) {
+              try {
+                ioe.getClass().getConstructor(String.class, Throwable.class)
```

Comment:
our new DynMethods classes (copied from Parquet) may help here. If they don't that's an opportunity for improvement